src/D3MOracle.sol

```diff
@@ -0,0 +1,117 @@
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+interface VatLike {
+    function live() external view returns (uint256);
+}
+
+interface HubLike {
+    function culled(bytes32) external view returns (uint256);
+}
+
+contract D3MOracle {
+    // --- Auth ---
+    /**
+        @notice Maps address that have permission in the Pool.
+        @dev 1 = allowed, 0 = no permission
+        @return authorization 1 or 0
+    */
+    mapping (address => uint256) public wards;
+    address public hub;
+
+    VatLike public immutable vat;
+    bytes32 public immutable ilk;
+
+    uint256 internal constant WAD = 10 ** 18;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, address data);
+
+    constructor(address vat_, bytes32 ilk_) {
+        vat = VatLike(vat_);
+        ilk = ilk_;
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /// @notice Modifier will revoke if msg.sender is not authorized.
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MOracle/not-authorized");
+        _;
+    }
+
+    // --- Administration ---
+    /**
+        @notice Makes an address authorized to perform auth'ed functions.
+        @dev msg.sender must be authorized.
+        @param usr address to be authorized
+    */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+        @notice De-authorizes an address from performing auth'ed functions.
+        @dev msg.sender must be authorized.
+        @param usr address to be de-authorized
+    */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+        @notice update vow or end addresses.
+        @dev msg.sender must be authorized.
+        @param what name of what we are updating bytes32("vow"|"end")
+        @param data address we are setting it to
+    */
+    function file(bytes32 what, address data) external auth {
+        require(vat.live() == 1, "D3MOracle/no-file-during-shutdown");
+
+        if (what == "hub") hub = data;
+        else revert("D3MOracle/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    /**
+        @notice Return value and status of the oracle
+        @return val always 1 WAD
+        @return ok true if vat is live or ilk is not culled
+    */
+    function peek() public view returns (uint256 val, bool ok) {
+        val = WAD;
+        ok = vat.live() == 1 || HubLike(hub).culled(ilk) == 0;
+    }
+
+    /**
+        @notice Return value
+        @dev vat must be live or ilk must be unculled in hub.
+        @return val always 1 WAD value
+    */
+    function read() external view returns (uint256 val) {
+        bool ok;
+        (val, ok) = peek();
+        require(ok, "D3MOracle/ilk-is-culled");
+    }
+}
```

Comment:
We are removing the ds-value's `void()` option (to set `ok` to 0). I don't see a problem with that (price is constant, no liquidations, in ES we don't want gov intervention anyhow), but pointing it in case someone sees a problem.

Reply:
Yeah it should be fine, it is kind of a "hardcoded" feed price.

Reply:
I suppose the DSValue is so different than other oracle implementations that preserving the interface is already a moot topic, but not having `void()`, not using storage variables for `has` and `val`, just means we are absolutely going to need to establish a new `class` pattern for these in the exec lib.  I bring this up just in case someone else can think of an edge case where this is a blocker.  I can't.

Reply:
Well this is an oracle that doesn't take more configuration than the `hub`, so there is not `poke` or `void`, just reading functions.

Reply:
A collateral already can have a `DSValue` an `Osm`, a `LPOsm` (with the different variants) and could even have directly a `Median`. The only real shared interface between all of them are `read` and `peek`.